### ISSUE TITLE
FOME console 92 instances of warning: [deprecation]

### ISSUE THREAD

**Issue by mi-hol** [FOME-Tech/fome-fw#197]
updated:

impact analysis in attached Spreadsheet
[warning_deprecation.ods](https://github.com/FOME-Tech/fome-fw/files/12071074/warning_deprecation.ods)

**Comment by mi-hol** [comment]
Affected are;

- ANTLRInputStream in org.antlr.v4.runtime
- FileLog in com.rusefi
- isSupportUpperBoundInclusive() in RealDistribution
- NotNull in org.antlr.v4.runtime.misc
- RandomData in org.apache.commons.math3.random
- RandomDataImpl in org.apache.commons.math3.random

**Comment by mi-hol** [comment]
> * NotNull in org.antlr.v4.runtime.misc

Reason for this seems usage of ANTLR (ANother Tool for Language Recognition) V 4.5 released Jan 23, 2015.
The current version is V  [4.13.0](https://github.com/antlr/antlr4/releases/tag/4.13.0) released May 21, 2023

Affected files are:

/java_console/ui/src/main/java/com/rusefi/livedata/generated/CPP14ParserBaseListener.java
/java_console/ui/src/main/java/com/rusefi/livedata/generated/CPP14ParserListener.java


they are both:

// Generated from CPP14Parser.g4 by ANTLR 4.5

unclear yet how to fix this issue because the only content of readme.md is "See https://github.com/antlr/grammars-v4/tree/master/cpp"
Would anyone have a hint?

**Comment by mi-hol** [comment]
> * RandomData in org.apache.commons.math3.random
> * RandomDataImpl in org.apache.commons.math3.random

Reason seems that source files from library org.apache.commons.math3.random have been directly copied into RE source tree

**Comment by mck1117** [comment]
Apache commons math lib fixed in 67decbdd4cf78447f26c5da42fd7cd379b601079

**Comment by mi-hol** [comment]
now the new gradle seems to be the only source for deprecation warning :)

```
Configure project :ui
The AbstractArchiveTask.destinationDir property has been deprecated. This is scheduled to be removed in Gradle 8.0. Please use the destinationDirectory property instead. See https://docs.gradle.org/7.5.1/dsl/org.gradle.api.tasks.bundling.AbstractArchiveTask.html#org.gradle.api.tasks.bundling.AbstractArchiveTask:destinationDir for more details.
	at build_arh196x3twauwl53158hq[23](https://github.com/FOME-Tech/fome-fw/actions/runs/5607176588/jobs/10258082196#step:7:24)la$_run_closure3.doCall(/home/runner/work/fome-fw/fome-fw/java_console/ui/build.gradle:28)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
```

**Comment by mck1117** [comment]
> now the new gradle seems to be the only source for deprecation warning :)

run `./gradlew clean` then build and you'll see all the warnings for every file

**Comment by mi-hol** [comment]
> run `./gradlew clean` then build and you'll see all the warnings for every file

I lack the permission to perform this on the level org FOME builds, should this be added to the workflow definition?

**Comment by mck1117** [comment]
all github actions happen in a fresh container - you only need that if building locally and have stale build intermediates

**Comment by mck1117** [comment]
that's strange, I get warnings in a local build, but can't see them in the GHA builds

![image](https://github.com/FOME-Tech/fome-fw/assets/568254/a8e792fe-60d4-4959-b0ac-33ea5768b48e)

**Comment by mi-hol** [comment]
correct, I looked up [this](https://github.com/FOME-Tech/fome-fw/actions/runs/5607176588/jobs/10258082196#step:7:24)  official  FOME build (not local build) and saw no warnings

**Comment by mck1117** [comment]
by the way, the same builds that run here are running in your fork: https://github.com/mi-hol/fome-fw/actions

**Comment by mi-hol** [comment]
> he same builds that run here are running in your fork

yes, I enabled GHA in my fork to test workflow changes before opening a PR. Do you see issues with that?

**Comment by mck1117** [comment]
Nope, that's completely fine, and in fact recommended. Your builds are completely separate from FOME, don't have access to FOME build secrets, etc.

**Comment by mi-hol** [comment]
> strange, I get warnings in a local build, but can't see them in the GHA builds

just to be sure, will you look into fixing this or shall I learn gradle?

**Comment by mi-hol** [comment]
> Apache commons math lib fixed in [67decbd](https://github.com/FOME-Tech/fome-fw/commit/67decbdd4cf78447f26c5da42fd7cd379b601079)

@mck1117 as I'm neither a JAVA nor Gradle expert, I wonder if  maybe the same approach would work for [com.devexperts.util](https://docs.dxfeed.com/dxlib/api/com/devexperts/util/package-summary.html) ?
It seems a clone of this code exists in fome-fw/java_console/logging-api/src/main/java/com/devexperts/util/

**Comment by mi-hol** [comment]
newest version of com.devexperts.logging.Logging is maintained on https://github.com/devexperts/QD/tree/master/dxlib

**Comment by mi-hol** [comment]
fixed, no more search hits for https://github.com/FOME-Tech/fome-fw/actions?query=workflow%3A%22%22FOME+console%22%22+depreca


Grandle 9 obsoletes again something, see https://github.com/FOME-Tech/fome-fw/actions/runs/12606851461/job/35137498291
